src/node/routes/index.ts

```diff
@@ -47,6 +49,9 @@ export const register = async (app: App, args: DefaultedArgs): Promise<Disposabl
   app.router.use(cookieParser())
   app.wsRouter.use(cookieParser())
 
+  const settings = new SettingsProvider<CoderSettings>(path.join(args["user-data-dir"], "coder.json"))
+  const updater = new UpdateProvider("https://api.github.com/repos/cdr/code-server/releases/latest", settings)
```

Comment:
This looks fancy 👀 So it takes in a URL to a release and a SettingProvider and returns an updater?
```suggestion
  const updater = new UpdateProvider("https://api.github.com/repos/coder/code-server/releases/latest", settings)
```

Reply:
Ooo nice catch.  Yup that is right.  The settings provider is used to store the last update info.

Reply:
Ah! I see. That makes sense

Reply:
Technically the updater only checks for updates and does not perform updates so the term `updater` might be inaccurate but I was not sure what else to call it. `updateChecker` maybe but it feels a bit long.